### PR TITLE
#4 implement admin user authentication

### PR DESCRIPTION
**Pull request by Sontsevolosa**
* Adding before_action to Application controller
* Making passing for old tests
* Adding a new test for verifying of redirect on index page when user is non logged in

### CODE REVIEW

**Comment by Sontsevolosa** [comment]
@mluts please, review my code

File: config/routes.rb [hunk 1]
```diff
@@ -1,6 +1,5 @@
 Rails.application.routes.draw do
   devise_for :users
-
```
Comment (mluts) [inline config/routes.rb]: Please put the newline back here

File: test/controllers/lunches_controller_test.rb [hunk 2]
```diff
@@ -1,9 +1,15 @@
...
 class LunchesControllerTest < ActionController::TestCase
-  test 'should get index' do
+  test 'should get index after successful login' do
+    sign_in users(:default)
     get :index
     assert_response :success
     assert_select 'title', 'Lunchez'
   end
+
+  test 'should not get index without successful login' do
```
Comment (mluts) [inline test/controllers/lunches_controller_test.rb]: @Sontsevolosa Could you please also write a test with `assert_redirected_to` ?

File: config/routes.rb (hunk 1 above)
Comment (Sontsevolosa) [inline config/routes.rb]: @mluts I deleted this line because Rubocop told me to do it :)

**Comment by Sontsevolosa** [comment]
@mluts  please, review my updated code

File: test/controllers/lunches_controller_test.rb (hunk 2 above)
Comment (Sontsevolosa) [inline test/controllers/lunches_controller_test.rb]: @mluts do you mean another test or including assert_redirected_to to existing test?

File: config/routes.rb (hunk 1 above)
Comment (mluts) [inline config/routes.rb]: @Sontsevolosa ok then

File: test/test_helper.rb [hunk 3]
```diff
@@ -10,3 +10,11 @@ class ActiveSupport::TestCase
 
   # Add more helper methods to be used by all tests here...
 end
+
+class ActionController::TestCase
+  include Devise::Test::ControllerHelpers
```
Comment (mluts) [inline test/test_helper.rb]: @Sontsevolosa 👍

File: test/controllers/lunches_controller_test.rb (hunk 2 above)
Comment (mluts) [inline test/controllers/lunches_controller_test.rb]: @Sontsevolosa another test

**Comment by mluts** [comment]
@Sontsevolosa reviewed, please consider my comments

**Comment by Sontsevolosa** [comment]
@mluts please, review my changes

File: test/controllers/lunches_controller_test.rb [hunk 4]
```diff
@@ -1,9 +1,20 @@
...
+    get :index
+    assert_response :redirect
+  end
+
+  test 'should redirect when user is non logged in' do
```
Comment (mluts) [inline test/controllers/lunches_controller_test.rb]: @Sontsevolosa please clarify test description

**Comment by Sontsevolosa** [comment]
@mluts please, review my code

**Comment by mluts** [comment]
@Sontsevolosa Good, please merge after CI green

**Pull request merged**